docs/source/download.rst

```diff
@@ -1,111 +1,134 @@
 
+   pip install devito
+   # ...or to install additional dependencies:
+   pip install devito[extras,mpi,nvidia]
+
+To install the latest Devito development version (current GitHub master) (without the tutorials), follow:
 
 .. code-block:: shell
 
-   pip install --user git+https://github.com/devitocodes/devito.git
+   pip install git+https://github.com/devitocodes/devito.git
+   # ...or to install additional dependencies:
+   pip install git+https://github.com/devitocodes/devito.git#egg=project[extras,mpi,nvidia]
+
+Additional dependencies:
+
+- extras : optional dependencies for Jupyter notebooks, plotting, benchmarking
+- mpi : optional dependencies for MPI (mpi4py)
+- nvidia : optional dependencies for targetting GPU deployment
 
+.. _pip package: https://pypi.org/project/devito/
 
-to install. Note that you don't need to get the code
-using git first in this method. 
+Note that you do not need to get the code via `git clone` in this method. 
 Depending on your needs, this might also be the recommended setup for using Devito
-in a production-like environment. However, since some of the components need to be
+in a production-like environment. However, since some components need to be
 compiled before use, this approach may be sensitive to the C/C++ compilers present
-on your system and the related environment including what other packages you might
+on your system and the related environment, including what other packages you might
 have installed.
 
-If you are facing any issues, we are happy to help on Slack_. Also look at our list of known
-installation issues_.
 
-Conda Environment
------------------
-If your objective is to contribute to and develop for Devito, the recommended way would
-be to use the included conda environment that also installs an appropriate C compiler
-along with all the bells and whistles we felt were necessary when developing for Devito.
-Please install either Anaconda_ or Miniconda_ using the instructions
-provided at the download links. You will need the Python 3 version.
+conda route
+```````````
+Please install either Anaconda_ or Miniconda_.
+
+.. _Anaconda: https://www.continuum.io/downloads
+
+.. _Miniconda: https://conda.io/miniconda.html
+
+.. _Python virtual environment: https://docs.python.org/3/library/venv.html
+
+.. _Conda environment: https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/environments.html
+
+.. code-block:: shell
+
+   # Create new env with the name devito
+   conda create --name devito
+   # Activate the environment
+   conda activate devito
+
+and finally, install Devito along with any extra dependencies:
+
+.. code-block:: shell
+
+   pip install devito
+   # ... or to install additional dependencies
+   pip install devito[extras,mpi,nvidia]
+
 
-To install Devito, including examples, tests and tutorial notebooks,
-follow these:
+For developers
+``````````````
+To develop with Devito, you may follow a similar approach as in `venv route`_ and `conda route`_
```

Comment:
Much less verbose...:

First of all, clone Devito:

...

and then install the requirements in your virtual environment (`venv` or `conda`):

...